pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -389,42 +398,69 @@ func (p *PoolManager) AddMacToWaitingConfig(allocations map[string]string) error
 	}
 
 	for _, macAddress := range allocations {
-		log.V(1).Info("add mac address to waiting config", "macAddress", macAddress)
-		macAddress = strings.Replace(macAddress, ":", "-", 5)
-		configMap.Data[macAddress] = time.Now().Format(time.RFC3339)
+		if isInRange, _ := p.IsMacInRange(macAddress); isInRange {
+			log.V(1).Info("add mac address to waiting config", "macAddress", macAddress)
+			macAddress = strings.Replace(macAddress, ":", "-", 5)
+			configMap.Data[macAddress] = time.Now().Format(time.RFC3339)
+		}
 	}
 
 	_, err = p.kubeClient.CoreV1().ConfigMaps(p.managerNamespace).Update(configMap)
 	return err
 }
 
 // Remove all the mac addresses from the waiting configmap this mean the vm was saved in the etcd and pass validations
-func (p *PoolManager) MarkVMAsReady(vm *kubevirt.VirtualMachine) error {
+func (p *PoolManager) MarkVMAsReady(vm *kubevirt.VirtualMachine) (error, bool) {
 	p.poolMutex.Lock()
 	defer p.poolMutex.Unlock()
 
 	configMap, err := p.kubeClient.CoreV1().ConfigMaps(p.managerNamespace).Get(vmWaitConfigMapName, metav1.GetOptions{})
 	if err != nil {
-		return err
+		return err, false
 	}
 
 	if configMap.Data == nil {
 		log.Info("the configMap is empty")
-		return nil
+		return nil, false
 	}
 
+	isMacAllocated := false
 	for _, vmInterface := range vm.Spec.Template.Spec.Domain.Devices.Interfaces {
-		if vmInterface.MacAddress != "" {
+		if isInRange, _ := p.IsMacInRange(vmInterface.MacAddress); vmInterface.MacAddress != "" && isInRange {
```

Comment:
sorry if I am to peaky, fill free to ignore.
isn't it worth to check if mac is empty, before we check if it is in range?
spare the function call...

Reply:
totally agree. I love picky reviews. please be as picky as you possible can :)